### PR TITLE
strcmp: change all !str[n]cmp to str[n]cmp == 0

### PR DESCRIPTION
**Pull request by sahlberg**
Change all !str[n]cmp to be str[n]cmp == 0  consistently across the codebase.

Issue #231   https://github.com/grpc/grpc/issues/231

Signed-off-by: Ronnie Sahlberg <ronniesahlberg@gmail.com>

### CODE REVIEW

**Comment by sahlberg** [comment]
Travis failed the python test.
I do not think it is my changes in this pull request but rather another instance of :
https://github.com/grpc/grpc/issues/966


Travis failed with :
...
TypeError: __init__() got an unexpected keyword argument 'syntax'

File: third_party/protobuf [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 644a6a1da71385e9d7a7a26b3476c93fdd71788c
```
Comment (ctiller) [inline third_party/protobuf]: You probably want to revert this change from this pull request: it's very unrelated.

Comment (nicolasnoble) [inline third_party/protobuf]: Agreed. This is what's causing travis to fail.

File: src/core/security/server_secure_chttp2.c [hunk 2]
```diff
@@ -139,11 +139,11 @@ int grpc_server_add_secure_http2_port(grpc_server *server, const char *addr, grp
...
   if (creds == NULL) goto error;
 
-  if (!strcmp(creds->type, GRPC_CREDENTIALS_TYPE_SSL)) {
+  if (strcmp(creds->type, GRPC_CREDENTIALS_TYPE_SSL) == 0) {
     status = grpc_ssl_server_security_context_create(
         grpc_ssl_server_credentials_get_config(creds), &ctx);
-  } else if (!strcmp(creds->type,
-                     GRPC_CREDENTIALS_TYPE_FAKE_TRANSPORT_SECURITY)) {
+  } else if (strcmp(creds->type,
+		    GRPC_CREDENTIALS_TYPE_FAKE_TRANSPORT_SECURITY) == 0) {
```
Comment (nicolasnoble) [inline src/core/security/server_secure_chttp2.c]: The repository's codestyle is using spaces instead of tabs.
